### PR TITLE
FDEM permittivity

### PR DESCRIPTION
**Pull request by lheagy**
#### Summary

Add the ability to perform forward simulations in the frequency domain with dielectric permittivity. Note that inversions are not yet supported: 
- derivatives are not yet implemented for the system when permittivity is non-zero (e.g. the gradients still all assume that dielectric permittivity = 0) -- this likely isn't a big deal as permittivity would have to be massive to cause a substantive change in the gradient, but this is something we can address in the future
- right now, `permittivity` is not an invertible property (something for the future) 

#### PR Checklist
* [x] If this is a work in progress PR, set as a Draft PR
* [x] Linted my code according to the [style guides](https://docs.simpeg.xyz/content/basic/practices.html#style).
* [x] Added [tests](https://docs.simpeg.xyz/content/basic/practices.html#testing) to verify changes to the code.
* [x] Added necessary documentation to any new functions/classes following the
      expect [style](https://docs.simpeg.xyz/content/basic/practices.html#documentation).
* [x] Added relevant method tags (i.e. `GRAV`, `EM`, etc.)
* [x] Marked as ready for review (ff this is was a draft PR), and converted 
      to a Pull Request
* [x] Tagged ``@simpeg/simpeg-developers`` when ready for review.

#### Reference issue

N/A

#### What does this implement/fix?

This extends the forward simulation capabilities and allows us to simulate complex conductivity / dielectric permittivity

#### Additional information

Note that the naming choice of `permittivity` rather than `epsilon` is different than the convention of naming properties by their greek letters and is in anticipation of the changes we will make with #1227

### CODE REVIEW

**Comment by lheagy** [comment]
Also, if you would like to try this out, there is an example here: https://github.com/lheagy/fdem-dielectric-permittivity

File: SimPEG/electromagnetics/frequency_domain/fields.py [hunk 1]
```diff
@@ -827,13 +836,16 @@ def _j(self, bSolution, source_list):
...
-            s_e = src.s_e(self.simulation)
-            j[:, i] = j[:, i] - s_e
+            for i, src in enumerate(source_list):
+                s_e = src.s_e(self.simulation)
+                j[:, i] = j[:, i] - s_e
 
-        return self._MeI * j
+            return self._MeI * j
+        else:
+            return self._MeI * self._MeSigma * self._e(bSolution, source_list)
```
Comment (jcapriot) [inline SimPEG/electromagnetics/frequency_domain/fields.py:848]:
```suggestion
            return self._MeI * (self._MeSigma * self._e(bSolution, source_list))
```
avoid the matrix-matrix multiplication

File: SimPEG/electromagnetics/frequency_domain/simulation.py [hunk 2]
```diff
@@ -52,10 +56,19 @@ class BaseFDEMSimulation(BaseEMSimulation):
...
 
-    def __init__(self, mesh, survey=None, forward_only=False, **kwargs):
+    def __init__(
+        self, mesh, survey=None, forward_only=False, permittivity=None, **kwargs
+    ):
         super().__init__(mesh=mesh, survey=survey, **kwargs)
         self.forward_only = forward_only
+        if permittivity is not None:
+            warnings.warn(
+                "Simulations using permittivity have not yet been thoroughly tested yet, and derivatives are not implemented. Contributions welcome!"
```
Comment (jcapriot) [inline SimPEG/electromagnetics/frequency_domain/simulation.py:69]:
```suggestion
                "Simulations using permittivity have not yet been thoroughly tested, and derivatives are not implemented. Contributions welcome!"
```

File: SimPEG/electromagnetics/frequency_domain/simulation.py [hunk 3]
```diff
@@ -552,6 +613,7 @@ def getRHSDeriv(self, freq, src, v, adjoint=False):
 ###############################################################################
 
 
+@with_property_mass_matrices("permittivity")
```
Comment (jcapriot) [inline SimPEG/electromagnetics/frequency_domain/simulation.py:616]:
```suggestion
```

You should be able to remove this as you are not doing anything like `MePermitivity`

**Review by jcapriot — CHANGES_REQUESTED** [review]
looks good, just a few small changes.

**Comment by lheagy** [comment]
accepted your changes, thanks Joe!

**Comment by lheagy** [comment]
Assuming this passes :) it is good to go from my perspective. Feel free to squash and merge. There are a number of small commits for fixing flake and small typos

**Comment by jcapriot** [comment]
I just tagged a new patch version for geoana, (0.4.1). Once that gets release we can update the version required for SimPEG to that, and it should fix the broken tests here.

**Pull request merged**